### ISSUE TITLE
rowEdit cancel not working as expected with row validation

### ISSUE THREAD

**Issue by ddaribo** [IgniteUI/igniteui-angular#12189]
## Description  
When the `rowEdit` event is canceled, the row's new data is cleared, however the cells' edit values remain. This is not related to the validation feature and is visible in the ["Event cancelation"](https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/editing#event-cancelation) demo in the docs:

![rowEdit-cancel-clel-editValues-remain](https://user-images.githubusercontent.com/35433937/195616884-b4bb8b8e-5745-4e67-945b-f28cc4502d53.gif)

In a grid validation scenario, it is expected that canceling the `rowEdit` event prevents the user from adding/editing an invalid row.
However, due to the above behavior, the grid accepts an invalid row.

To an extent related to #12127.

 * igniteui-angular version: latest (& any)
 * browser: any

## Steps to reproduce  

1. Open [this sample](https://stackblitz.com/edit/igx-grid-validation-cancel-rowedit?file=src/app/sample/igx-sample/igx-sample.component.html)
2. Observe that the `ProductName` and `CategoryName` fields are required
3. Click the "Add" button on the action strip to add a new row
4. Type something only in the `ProductName` field
5. Click `Done` in attempt to finalize the row edit while the `CategoryName` field is left empty
6. You will see the error message and the `ProductName` cell's value reset. (At this point, if you double click on `ProductName` you will see its edit value remains as before clicking `Done`)
8. Fill a value in `CategoryName`
9. Click `Done`

## Result  
The row is added, however, with an invalid state, event though the `rowEdit` event is canceled with the intention not to add/edit an invalid row:
```
public rowEdit(args: IGridEditEventArgs) {
    if (!args.valid) {
      args.cancel = true;
    }
  }
```

## Expected result  
Reset the cell edit values to the cells' previous values when row editing is canceled?
Properly recalculate the row's validity state.

## Attachments

**Comment by mmart1n** [comment]
Also, when the `cell editing` is canceled via `Escape`, the `cell.editValue` is not reverted to the `cell.value`.

![cell edit cancel](https://user-images.githubusercontent.com/44684385/208678675-f1ec995b-9115-419b-8248-00811873911e.gif)

**Comment by mddragnev** [comment]
@ddaribo It was discussed that even normal (without validation) rowEdit cancelation should not clear any values you entered during cell updating because it is not like canceling the rowEdit with escape or cancel button. So in the mentioned PR I changed that and made it consistent between row adding and row editing. Therefore with this change the issue should be fixed.
Also, I have fixed the additional issue that @mmart1n mentioned above.
